SuperCalculator/src/supercalculator/Power.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Power class
+ *
+ * August 9th 2021
+ *
+ */
+
+package supercalculator;
+
+import java.util.Scanner;
+
+/**
+ * Power function for super calculator SOEN 6011
+ */
+public class Power {
+
+    /**
+    * Default constructor
+    */
+    public Power() { }
+
+    /**
+     * Calculates the result of the base elevated to the given power using the binary exponentiation algorithm
+     * @param base The number that is the base for the operation
+     * @param power The number to which the base is elevated to
+     */
+    public double calculate(double base, int power) {
+        /* TODO: Also must add unit tests */
+        double result = 1; // Holds the result over the steps
+        double b = base; // Keeps control of base multiplications
+
+        /* 0 elevated to any number greater than zero is 0 */
+        if(base == 0) {
+            return 0;
+        }
+
+        /* Any number to the power of 0 is 1 */
+        if(power == 0) {
+            return 1;
+        }
+
+        while (power > 0) {
+            if((power & 1) == 1) {
+                result = result * b;
+            }
+
+            b = b * b;
+            power >>= 1;
+        }
+
+        return result;
+    }
+
+    /**
+     * Handles user input and output
+     */
+    public void getInput() {
+        Scanner scanner = new Scanner(System.in);
```

Comment:
The team agreed to use only one Stream for the app as we were having an exception thrown when closing the stream in a particular function. The way we handled this was to pass the scanner to the getInput(scanner) to avoid instantiating the scanner directly in the function. 

See how other functions implemented it. e.g. Tan or Beta